src/commands/aksCRUDDiagnostics/aksCRUDDiagnostics.ts

```diff
@@ -39,19 +39,27 @@ async function loadDetector(
     await longRunning(`Loading ${clustername} diagnostics.`,
           async () => {
             const clusterAppLensData = await getAppLensDetectorData(cloudTarget, "aks-category-crud");
-            const detectorMap = new Map();
+            if (clusterAppLensData === undefined) {
```

Comment:
Moving this check to the start (it was at the end before) as we'll never complete this command if `clusterAppLensData` is not defined.

Also I like to have an explicit error message even (or especially) if this is never expected to happen.